app/src/main/java/com/google/samples/apps/sunflower/compose/home/HomeScreen.kt

```diff
@@ -0,0 +1,163 @@
+import com.google.samples.apps.sunflower.compose.garden.GardenScreen
+import com.google.samples.apps.sunflower.compose.plantlist.PlantListScreen
+import com.google.samples.apps.sunflower.data.Plant
+import kotlinx.coroutines.launch
+
+enum class SunflowerPage(
+    @StringRes val titleResId: Int,
+    @DrawableRes val drawableResId: Int
+) {
+    MY_GARDEN(R.string.my_garden_title, R.drawable.ic_my_garden_active),
+    PLANT_LIST(R.string.plant_list_title, R.drawable.ic_plant_list_active)
+}
+
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+fun HomeScreen(onPlantClick: (Plant) -> Unit, onPageChange: (SunflowerPage) -> Unit) {
+    val pagerState = rememberPagerState()
+
+    LaunchedEffect(pagerState.currentPage) {
+        onPageChange(SunflowerPage.values()[pagerState.currentPage])
+    }
+
+    Column(Modifier.nestedScroll(rememberNestedScrollInteropConnection())) {
+        val coroutineScope = rememberCoroutineScope()
+
+        // Tab Row
+        TabRow(selectedTabIndex = pagerState.currentPage) {
+            SunflowerPage.values().forEachIndexed { index, page ->
+                val title = stringResource(id = page.titleResId)
+                Tab(
+                    selected = pagerState.currentPage == index,
+                    onClick = { coroutineScope.launch { pagerState.scrollToPage(index) } },
+                    text = { Text(text = title) },
+                    icon = {
+                        Icon(
+                            painter = painterResource(id = page.drawableResId),
+                            contentDescription = title
+                        )
+                    },
+                    unselectedContentColor = MaterialTheme.colors.primaryVariant,
+                    selectedContentColor = MaterialTheme.colors.secondary,
+                )
+            }
+        }
+
+        // Pages
+        HorizontalPager(
+            pageCount = SunflowerPage.values().size,
+            state = pagerState,
+            verticalAlignment = Alignment.Top
+        ) { index ->
+            when (SunflowerPage.values()[index]) {
+                SunflowerPage.MY_GARDEN -> {
+                    GardenScreen(
+                        Modifier.fillMaxSize(),
+                        onAddPlantClick = {
+                            coroutineScope.launch {
+                                pagerState.scrollToPage(SunflowerPage.PLANT_LIST.ordinal)
+                            }
+                        }, onPlantClick = {
+                            onPlantClick(it.plant)
+                        })
+                }
+                SunflowerPage.PLANT_LIST -> {
+                    PlantListScreen(
+                        onPlantClick = onPlantClick,
+                        Modifier.fillMaxSize(),
+                    )
+                }
+            }
+        }
+    }
+}
+
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+private fun HomeTopAppBar(pagerState: PagerState, onFilterClick: () -> Unit) {
```

Comment:
This is unused since I couldn't use `Scaffold` because of the M2 collapsing toolbar implementation. This will be useful though for when Sunflower is migrated to M3 in #829